src/cascade/model/session.py

```diff
@@ -0,0 +1,552 @@
+        # else use the one generated by the call to init, coming from the mean.
+        self._run_dismod(["fit", fit_level])
+        return FitResult(self, self.get_var("fit"))
+
+    def _setup_model(self, model, data):
+        extremal = list()
+        if data is not None and not data.empty:
+            for dimension in ["age", "time"]:
+                cols = [ac for ac in data.columns if ac.startswith(dimension)]
+                if not cols:
+                    raise ValueError(f"Dataframe must have age and time columns but has {data.columns}.")
+                extremal.append({data[cols].min().min(), data[cols].max().max()})
+        self.write_model(model, extremal)
+        self.write_data(data)
+        self._run_dismod(["init"])
+        if model.scale_set_by_user:
+            self.set_var("scale", model.scale)
+        else:
+            # Assign to the private variable because setting the property
+            # indicates that the user of the API wants to set their own scale
+            # instead of using the one Dismod-AT calculates during init.
+            model._scale = self.get_var("scale")
+
+    def predict(self, var, avgint, parent_location, weights=None, covariates=None):
+        """Given rates, calculated the requested average integrands.
+
+        Args:
+            var (DismodGroups): Var objects with rates.
+            avgint (pd.DataFrame): Request data in these ages, times, and
+                locations. Columns are ``integrand`` (str), ``location``
+                (location_id), ``age_lower`` (float), ``age_upper`` (float),
+                ``time_lower`` (float), ``time_upper`` (float). The integrand
+                should be one of the names in IntegrandEnum.
+            parent_location: The id of the parent location.
+            weights (Dict[Var]): Weights are estimates of ``susceptible``,
+                ``with_condition``, and ``total`` populations, used to bias
+                integrands with age or time extent. Each one is a single
+                Var object.
+            covariates (List[Covariate]): A list of Covariates, so that we know
+                the name and reference value for each.
+
+        Returns:
+            (pd.DataFrame, pd.DataFrame): The predicted avgints, and a dataframe
+            of those not predicted because their covariates are greater than
+            ``max_difference`` from the ``reference`` covariate value.
+            Columns in the ``predicted`` are ``predict_id``, ``sample_index``,
+            ``avg_integrand`` (this is the value), ``location``, ``integrand``,
+            ``age_lower``, ``age_upper``, ``time_lower``, ``time_upper``.
+        """
+        self._check_vars(var)
+        model = Model.from_var(var, parent_location, weights=weights, covariates=covariates)
+        extremal = ({avgint.age_lower.min(), avgint.age_upper.max()},
+                    {avgint.time_lower.min(), avgint.time_upper.max()})
+        self.write_model(model, extremal)
```

Comment:
This is not using the _setup_model method because that is only for fitting? Can we make that more general so it can be used in this case as well. Or make the name more specific so it's clear why it's not used here.

Reply:
Renamed to setup_model_for_fit. When fitting, we should be setting the start var and scale var. This does that. When doing predict, we should be setting the truth var, and that's what you see in this code. They are separate setups.

I renamed this to setup_model_for_fit.